python-frontend/src/test/java/org/sonar/python/tree/IPythonTreeMakerTest.java

```diff
@@ -0,0 +1,155 @@
+    assertThat(parse).isNotNull();
+    lineMagic = findFirstChildOf(parse, Tree.Kind.LINE_MAGIC);
+    assertThat(lineMagic)
+      .isNotNull()
+      .isInstanceOf(LineMagic.class);
+
+    parse = parseIPython("print(b)\n" +
+      "a = %timeit foo(b) % 3\n" +
+      "print(a)", treeMaker::fileInput);
+    assertThat(parse).isNotNull();
+    lineMagic = findFirstChildOf(parse, Tree.Kind.LINE_MAGIC);
+    assertThat(lineMagic)
+      .isNotNull()
+      .isInstanceOf(LineMagic.class);
+  }
+
+  @Test
+  public void line_magic_statement() {
+    var parse = parseIPython("print(b)\n" +
+      "%alias showPath pwd && ls -a\n", treeMaker::fileInput);
+    assertThat(parse).isNotNull();
+    var lineMagicStatement = findFirstChildOf(parse, Tree.Kind.LINE_MAGIC_STATEMENT);
+    assertThat(lineMagicStatement)
+      .isNotNull()
+      .isInstanceOf(LineMagicStatement.class);
+
+    assertThat(lineMagicStatement.children()).hasSize(1);
+    var lineMagic = findFirstChildOf(lineMagicStatement, Tree.Kind.LINE_MAGIC);
+    assertThat(lineMagic).isNotNull();
+
+    parse = parseIPython("print(b)\n" +
+      "%timeit a = foo(b) % 3\n" +
+      "a %= 2\n" +
+      "print(a)", treeMaker::fileInput);
+    assertThat(parse).isNotNull();
+    lineMagicStatement = findFirstChildOf(parse, Tree.Kind.LINE_MAGIC_STATEMENT);
+    assertThat(lineMagicStatement)
+      .isNotNull()
+      .isInstanceOf(LineMagicStatement.class);
+
+    assertThat(lineMagicStatement.children()).hasSize(1);
+    lineMagic = findFirstChildOf(lineMagicStatement, Tree.Kind.LINE_MAGIC);
+    assertThat(lineMagic).isNotNull();
+
+    parse = parseIPython("print(b)\n" +
+      "%timeit a = foo(b); b = 2\n" +
+      "a += b\n" +
+      "print(a)", treeMaker::fileInput);
+    assertThat(parse).isNotNull();
+    lineMagicStatement = findFirstChildOf(parse, Tree.Kind.LINE_MAGIC_STATEMENT);
+    assertThat(lineMagicStatement)
+      .isNotNull()
+      .isInstanceOf(LineMagicStatement.class);
+
+    assertThat(lineMagicStatement.children()).hasSize(1);
+    lineMagic = findFirstChildOf(lineMagicStatement, Tree.Kind.LINE_MAGIC);
+    assertThat(lineMagic).isNotNull();
+
+    parse = parseIPython("print(b)\n" +
+      "%autocall 1\n", treeMaker::fileInput);
+    assertThat(parse).isNotNull();
+  }
+
+  @Test
+  public void assignment_rhs_test() {
+    var parse = parseIPython("print(b)\n" +
+      "a = yield foo(b)\n" +
+      "c = bar(a) + b", treeMaker::fileInput);
+    assertThat(parse).isNotNull();
+
+    var assignments = findChildrenOf(parse, Tree.Kind.ASSIGNMENT_STMT);
+    assertThat(assignments).hasSize(2);
+    var yieldExpression = findFirstChildOf(assignments.get(0), Tree.Kind.YIELD_EXPR);
+    assertThat(yieldExpression).isNotNull();
+    var binaryExpression = findFirstChildOf(assignments.get(1), Tree.Kind.PLUS);
+    assertThat(binaryExpression).isNotNull();
+  }
+
+  private <T extends Tree> List<T> findChildrenOf(Tree parent, Tree.Kind kind) {
```

Comment:
Sorry I missed this during the first review.
I'm getting quite ugly warning from IntelliJ due to the "unchecked" cast. I know it's safe, but I also see we don't actually use the generic.
What about simply returning a  `List<Tree>` (and a `Tree` in the other method) to simplify things and get rid of the warning?
Also, I suggest renaming the methods `findChildrenWithKind` and `findFirstChildWithKind` to make it a bit clearer.

Reply:
done